src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -1069,4 +1029,184 @@ final case class EServicesApiServiceImpl(
+
+      val result = for {
+        mimeType    <- Try(new Tika().detect(readFileAsBytes(filePath), fileName)).toEither.toFuture
+        contentType <- ContentType
+          .parse(mimeType)
+          .leftMap(errors => ContentTypeParsingError(mimeType, file.path, errors.map(_.toString())))
+          .toFuture
+        fileParts = (FileInfo("", fileName, contentType), new File(filePath))
+        _ <- verifyAndCreateEServiceDocument(
+          catalogProcessService,
+          fileManager,
+          eService,
+          fileParts,
+          file.prettyName,
+          fileType,
+          descriptor.id,
+          uuidSupplier
+        )
+      } yield ()
+
+      result.recoverWith { case ex: Throwable =>
+        deleteTempDirectory(zipPath)
+        for {
+          _ <- descriptor.docs.traverse { doc =>
+            catalogProcessService.deleteEServiceDocumentById(eService.id, descriptor.id, doc.id)
+          }
+          _ <- catalogProcessService.deleteEService(eService.id)
+          _ <- catalogProcessService.deleteDraft(eService.id, descriptor.id)
+        } yield throw ex
+      }
+    }
+
+    val result: Future[CreatedResource] = for {
+      tenantId         <- getOrganizationIdFuture(contexts)
+      zipFile          <- fileManager.getFile(ApplicationConfiguration.importEServiceContainer)(
+        s"${ApplicationConfiguration.importEServicePath}/$tenantId/${fileResource.filename}"
+      )
+      zipPath          <- extractZipToTempDirectory(zipFile, tenantId).toFuture
+      importedEservice <- checkZipStructure(zipPath).toFuture.recoverWith { case ex: Throwable =>
+        deleteTempDirectory(zipPath)
+        throw ex
+      }
+      eserviceSeed = CatalogProcess.EServiceSeed(
+        name = importedEservice.name,
+        description = importedEservice.description,
+        technology = importedEservice.technology.toProcess,
+        mode = importedEservice.mode.toProcess
+      )
+      eService <- catalogProcessService.createEService(eserviceSeed).recoverWith { case ex: Throwable =>
+        deleteTempDirectory(zipPath)
+        throw ex
+      }
+      descriptorSeed = CatalogProcess.EServiceDescriptorSeed(
+        audience = importedEservice.descriptor.audience,
+        voucherLifespan = importedEservice.descriptor.voucherLifespan,
+        dailyCallsPerConsumer = importedEservice.descriptor.dailyCallsPerConsumer,
+        dailyCallsTotal = importedEservice.descriptor.dailyCallsTotal,
+        agreementApprovalPolicy = importedEservice.descriptor.agreementApprovalPolicy.toProcess,
+        attributes = CatalogProcess.AttributesSeed(certified = Seq.empty, declared = Seq.empty, verified = Seq.empty)
+      )
+      descriptor <- catalogProcessService.createDescriptor(eService.id, descriptorSeed).recoverWith {
+        case ex: Throwable =>
+          deleteTempDirectory(zipPath)
+          for {
+            _ <- catalogProcessService.deleteEService(eService.id)
+          } yield throw ex
```

Comment:
minor, we can avoid for comprehension here
```suggestion
          catalogProcessService.deleteEService(eService.id).flatMap(_ => Future.failed(ex))
```